.github/scripts/lintrunner.sh

```diff
@@ -24,15 +24,17 @@ lintrunner init 2> /dev/null
 if [[ "${CLANG}" == "1" ]]; then
     python3 -m tools.linter.clang_tidy.generate_build_files
 fi
-echo "Generating torch version"
-python3 -m tools.generate_torch_version --is_debug=false
-echo "Generating pyi files"
-python3 -m tools.pyi.gen_pyi \
-    --native-functions-path aten/src/ATen/native/native_functions.yaml \
-    --tags-path aten/src/ATen/native/tags.yaml \
-    --deprecated-functions-path "tools/autograd/deprecated.yaml"
-echo "Generating datapipes pyi files"
-python3 torch/utils/data/datapipes/gen_pyi.py
+if [[ "${MYPY}" == "1" ]]; then
+    echo "Generating torch version"
+    python3 -m tools.generate_torch_version --is_debug=false
+    echo "Generating pyi files"
+    python3 -m tools.pyi.gen_pyi \
+        --native-functions-path aten/src/ATen/native/native_functions.yaml \
+        --tags-path aten/src/ATen/native/tags.yaml \
+        --deprecated-functions-path "tools/autograd/deprecated.yaml"
+    echo "Generating datapipes pyi files"
+    python3 torch/utils/data/datapipes/gen_pyi.py
```

Comment:
Wow. This is pre-existing behavior, but it feels wrong that we're installing these things here instead of as part of the `init_command` for mypy in https://github.com/pytorch/pytorch/blob/main/.lintrunner.toml#L151

Any idea why these were put here? Seems like a way to guarantee CI behavior diverges from regular dev laptop behavior

Reply:
I genuinely don't know why. The first time I looked at this script was today.

Reply:
My guess is that these are things that would get generated from a normal pytorch build so generally you would have these things locally if you were doing actual dev.